script-reference/types.rst

```diff
@@ -520,15 +520,109 @@ Type Conversions
 pattern
 -------
 
-A type representing regular-expression patterns that can be used
-for fast text-searching operations.  Pattern constants are created
-by enclosing text within forward slashes (``/``) and use the same syntax
-as the patterns supported by the `flex lexical analyzer
-<http://westes.github.io/flex/manual/Patterns.html>`_.  The speed of
-regular expression matching does not depend on the complexity or
-size of the patterns.  Patterns support two types of matching, exact
-and embedded.
+A type representing regular-expression patterns that can be used for fast
+text-searching operations.  Pattern constants are created by enclosing text
+within forward slashes (``/``) and support a large subset of the `flex lexical
+analyzer <http://westes.github.io/flex/manual/Patterns.html>`_ syntax.  As in
+other implementations, patterns consist of ordinary and special characters.
+Patterns such as ``/a/`` or ``/A0123/`` match that specific character byte or
+character sequence, case-sensitively. Special characters and modifiers customize
+matching, as follows:
 
+.. list-table::
+  :header-rows: 1
+
+  * - Syntax
+    - Meaning
+
+  * - ``^``
+    - Matches the beginning of the input.
+
+  * - ``$``
+    - Matches the end of the input.
+
+  * - ``.``
+    - Matches any character except newline (but see the ``(?s:`` and ``/s`` modifiers).
+
+  * - ``<expr>*``
+    - Matches zero or more instances of ``expr``.
+
+  * - ``<expr>+``
+    - Matches one or more instances of ``expr``.
+
+  * - ``<expr>?``
+    - Matches zero or one instances of ``expr``.
+
+  * - ``<expr>{n}``
+    - Matches ``expr`` ``n`` times, where ``n`` is a non-negative integer.
+
+  * - ``<expr>{n,}``
+    - Matches ``expr`` ``n`` or more times, where ``n`` is a non-negative integer.
+
+  * - ``<expr>{n,m}``
+    - Matches ``expr`` between ``n`` and ``m`` times, inclusively, where ``n``
+      and ``m`` are non-negative integers and ``m >= n``.
+
+  * - ``<expr1>|<expr2>``
+    - Matches either ``expr1`` or ``expr2``. In contrast to the above repetition
+      operators, expressions here span multiple characters: in ``ab+`` the ``+``
+      applies to the ``b``, while ``ab|cd`` matches ``"ab"`` or ``"cd"``.
```

Comment:
I'd actually skip the 2nd sentence, find that more confusing than helpful.

```suggestion
    - Matches either ``expr1`` or ``expr2``. 
```

Reply:
I went back and forth over that one a bunch. :-) Happy to remove it.